### PR TITLE
Microsoft Login

### LINKED ISSUE #710: Microsoft account login
**Is your feature request related to a problem? Please describe.**
New Minecraft accounts are already getting created with Microsoft account, it would be nice if GDLauncher supported that

**Describe the solution you'd like**
A user in MultiMC/MultiMC5#3475 did some research and documented the process

### PR DESCRIPTION
**Pull request by EnzoAlbornoz**
## Purpose
Currently, GDLauncher cannot provide login for Microsoft accounts. As far we know, soon Mojang accounts will be migrated to Microsoft, then we need to be able to perform this type of login.

## Related Issues:
Closes #710 - Microsoft account login

## Approach
This feature will turn GDLauncher able to perform OAuth login for Microsoft accounts.

#### Open Questions and Pre-Merge TODOs
- [x] User able to perform Microsoft authentication on login page
- [x] Update Azure App ClientId of GDLauncher to an official one
- [x] User able to add new Microsoft accounts from home page
- [x] Auto renew of Microsoft tokens (Microsoft, XBL, XSTS, Minecraft)

## Learning
https://wiki.vg/Microsoft_Authentication_Scheme
https://github.com/OpenXbox/xbox-webapi-python
https://docs.microsoft.com/en-us/gaming/xbox-live/api-ref/xbox-live-rest/additional/edsauthorization
https://docs.microsoft.com/en-us/advertising/guides/authentication-oauth-live-connect?view=bingads-13
https://www.valentinog.com/blog/challenge/

### CODE REVIEW

**Comment by blarfoon** [comment]
Looking good so far. I'll give it a complete review tmr morning and eventually merge this.

File: src/common/api.js [hunk 1]
```diff
@@ -1,16 +1,114 @@
...
+  MICROSOFT_LIVE_LOGIN_URL,
+  MICROSOFT_XBOX_LOGIN_URL,
+  MICROSOFT_XSTS_AUTH_URL,
+  MINECRAFT_SERVICES_URL
 } from './utils/constants';
 import { sortByDate } from './utils';
 
+// Microsoft Auth
+
+export const msExchangeCodeForAcessToken = (
```
Comment (blarfoon) [inline src/common/api.js]: There's a typo here, it should be `msExchangeCodeForAccessToken `

File: src/common/reducers/actions.js [hunk 2]
```diff
@@ -486,6 +635,143 @@ export function loginThroughNativeLauncher() {
...
+        console.error(error);
+        throw new Error('Error occurred while logging in Microsoft.');
+      }
+
+      let msAccessToken = null;
+      let msRefreshToken = null;
+      try {
+        ({
+          data: { access_token: msAccessToken, refresh_token: msRefreshToken }
+        } = await msExchangeCodeForAcessToken(
```
Comment (blarfoon) [inline src/common/reducers/actions.js]: Typo here too

**Review by blarfoon — APPROVED** [review]
PR is good. Let's fix the typos and get this merged 🎉

File: src/common/api.js (hunk 1 above)
Comment (EnzoAlbornoz) [reply src/common/api.js]: Fixed.

File: src/common/reducers/actions.js (hunk 2 above)
Comment (EnzoAlbornoz) [reply src/common/reducers/actions.js]: Fixed

**Pull request merged**